drivers/can/Kconfig.mcp2515

```diff
@@ -0,0 +1,132 @@
+# Kconfig.mcp2515 - MCP2515 CAN configuration options
+
+#
+# Copyright (c) 2018 Karsten Koenig
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config CAN_MCP2515
+	bool "MCP2515 CAN Driver"
+	default n
+	depends on SPI
+	help
+	  Enable MCP2515 CAN Driver
+
+if CAN_MCP2515
+
+config CAN_MCP2515_NAME
+	string "Driver name"
+	default "MCP2515"
+
+config CAN_MCP2515_OSC_FREQ
+	int "Oscillator frequency"
+	default 8000000
+	help
+	 Specify the frequency of the oscillator connected to the MCP2515.
+
+config CAN_PROP_SEG
+	int "Prop_Seg"
+	default 2
+	range 1 8
+	help
+	  Time quantums of propagation segment (ISO 11898-1)
+
+config CAN_PHASE_SEG1
+	int "Phase_Seg1"
+	default 7
+	range 1 8
+	help
+	  Time quantums of phase buffer 1 segment (ISO 11898-1)
+
```

Comment:
I wanted to avoid doing any unexpected math, and there is hard limits on the values, but I think you are right, I could just do the following to allow for full range and stay in the bounds, will keep it in mind when switching to dts proper
tq_prop = tq_prop_bs1 / 2
tq_bs = (tq_prop_bs1 / 2) + (tq_prop_bs1 % 2)

Reply:
I had
tq_bs1 = (tq_prop_bs1 -1) & 0x07
tq_prop = tq_prop_bs1 - tq_bs1 - 2
in mind but its the same :-)